src/documents/tutorial-developerIntroduction/DeveloperIntroductionToInfusionFramework-RestructuringComponents.md

```diff
@@ -0,0 +1,116 @@
+
+In the example below, we extract the two "say hello" components into separate component definitions from the main component, then include them as subcomponents of the main component with their options appropriately configured. We'll also add a listener to the main component to announce (once) when its creation is complete.
+
+In the process, we'll be looking at some other important characteristics of Infusion component design:
+
+1. Building freestanding components, then wiring them together for more complex behaviours.
+
+2. Overriding default component configuration when building up applications from separate components.
+
+
+<div class="infusion-docs-note">You can check out the [Live Example](http://codepen.io/waharnum/pen/egBObY?editors=1111) of the code below on [CodePen](http://codepen.io/)</div>
+
+```
+// The console hello functionality is now defined as a separate
+// component
+fluid.defaults("fluid.helloWorld.consoleHello", {
+    gradeNames: ["fluid.modelComponent"],    
+    // We define a default message here so that
+    // this component is fully independent and
+    // could be used on its own
+    model: {
+        message: "Hello, Console World!"
+    },
+    modelListeners: {
+        "message": "{that}.sayHello"
+    },
+    invokers: {
+        sayHello: {
+            "this": "console",
+            "method": "log",
+            // Here, "{that}" means the context of the current
+            // component configuration (consoleHello)
+            "args": ["{that}.model.message"]
+        },
+    }
+});
+
+// The web page hello functionality is now defined as a separate
+// component
+fluid.defaults("fluid.helloWorld.displayHello", {
+    gradeNames: ["fluid.viewComponent"],
+    // We define a default message here so that
+    // this component is fully independent and
+    // could be used on its own
+    model: {
+        message: "Hello, Web Page World!"
+    },
+    selectors: {
+        messageArea: ".flc-messageArea"
+    },
+    modelListeners: {
+        "message": "{that}.displayHello"
+    },
+    invokers: {
+        displayHello: {
+            "this": "{that}.dom.messageArea",
+            "method": "html",
+            "args": ["{that}.model.message"]
+        }
+    }
+});
+
+fluid.defaults("fluid.helloWorld", {
+    gradeNames: ["fluid.modelComponent"],
+    model: {
+        message: "Hello, World!"
+    },
+    listeners: {
+        "onCreate.announceSelf": {
+            "this": "console",
+            "method": "log",
+            "args": ["The helloWorld Component is ready"]
+        }
+    },
+    components: {
+        consoleHello: {
+            type: "fluid.helloWorld.consoleHello",
+            options: {
+                model: {
+                    message: "{helloWorld}.model.message"
+                }
+            }
+        },
+        dispayHello: {
```

Comment:
The phenomenon described at https://www.wired.com/2014/08/wuwt-typos/ in action. Fixed.